Source/WebInspectorUI/UserInterface/Views/DOMTreeElement.js

```diff
@@ -1381,26 +1381,116 @@ WI.DOMTreeElement = class DOMTreeElement extends WI.TreeElement
         this._updateBreakpointStatus();
     }
 
+    _buildAccessibilityDOM(accessibilityNode, baseURL)
+    {
+        var domNode = accessibilityNode.domNode;
+        var titleDOM = document.createDocumentFragment();
+        var tagElement = document.createElement("span");
+        tagElement.className = "accessibility-tree-node";
+        if (accessibilityNode.isIgnored) 
+            tagElement.classList.add("disabled-element");
+        
+        var tagNameElement = document.createElement("span");
+        let roleSpan = document.createElement("span");
+        roleSpan.className = "accessibility-role";
+        roleSpan.textContent = accessibilityNode.computedRole;
+        tagNameElement.appendChild(roleSpan);
+        if (accessibilityNode.computedLabel) {
+            let colon = document.createTextNode(": ");
+            let labelSpan = document.createElement("span");
+            labelSpan.className = "accessibility-label";
+            labelSpan.textContent = `"${accessibilityNode.computedLabel}"`;
+            tagNameElement.appendChild(colon);
+            tagNameElement.appendChild(labelSpan);
+        }
+        if (typeof accessibilityNode.focused === "boolean") {
+            let focusSpan = document.createElement("span");
+            focusSpan.className = "accessibility-focus";
+            focusSpan.textContent = ` Focusable: ${accessibilityNode.focused}`;
+            tagNameElement.appendChild(focusSpan);
+        }
+        var attributes = domNode.attributes();
+        for (var i = 0; i < attributes.length; ++i) {
+            var attr = attributes[i];
+            let rewrittenURL = absoluteURL(attr.value, baseURL);
+            if (rewrittenURL &&  (attr.name === "src" || /\bhref\b/.test(attr.name))) {
+                let urlSpan = document.createElement("span");
+                urlSpan.className = "accessibility-base-url";
+                urlSpan.textContent = ` Url: ${rewrittenURL}`;
+                tagNameElement.appendChild(urlSpan);
+            }
+        }
+        this._addAccessibilityWarnings(tagNameElement, accessibilityNode);
+        tagElement.appendChild(tagNameElement);
+        titleDOM.appendChild(tagElement);
+        titleDOM.append("\u200B");
+        return titleDOM;
+    }
+
+    _addAccessibilityWarnings(tagNameElement, accessibilityNode)
+    {
+        if (accessibilityNode.hasWarnings) {
+            let warningSpan = document.createElement("span");
+            warningSpan.className = "accessibility-warning";
+            let warningIcon = document.createElement("img");
+            warningIcon.src = "Images/Warning.svg"; 
+            warningIcon.alt = "Warning: missing label";
+            warningIcon.className = "accessibility-warning-icon";
+            
+            // Use the first warning message as the tooltip
+            if (accessibilityNode.warningMessages.length > 0) {
+                warningIcon.title = accessibilityNode.warningMessages[0];
+            }
```

Comment:
If it possible to show more than one warning, it makes sense to display them all as a series of icons with their own title.

See this helper:
https://github.com/WebKit/WebKit/blob/f046841ed906537113e5652978a52364ea86e139/Source/WebInspectorUI/UserInterface/Views/RecordingActionTreeElement.js#L415